images/gce-dind/Dockerfile

```diff
@@ -0,0 +1,38 @@
+# Copyright 2018 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+FROM gcr.io/k8s-testimages/kubekins-e2e:v20180102-0e2b24a0b-master
+LABEL maintainer "leblancd@cisco.com"
+
+# add env we can debug with the image name:tag
+ARG IMAGE_ARG
+ENV IMAGE=${IMAGE_ARG}
+
+RUN apt-get update && apt-get install -y \
+    git \
+    wget \
+    liblz4-1 && \
+    apt-get clean
+
+# Install kubeadm-dind-cluster scripts
+RUN cd /root && \
+    git clone -b k8s_ipv6_ci https://github.com/leblancd/kubeadm-dind-cluster.git
+
+# Install docker-machine
+RUN curl -L https://github.com/docker/machine/releases/download/v0.13.0/docker-machine-`uname -s`-`uname -m` >/tmp/docker-machine && install /tmp/docker-machine /usr/local/bin/docker-machine
```

Comment:
Why do we need docker machine, and shouldn't this be managed by apt-get?

Reply:
The kubeadm-dind-cluster scripts use docker-machine to both create the instance on GCE, and to create a connection to the docker daemon running on that instance. Once the docker-machine connection is set up, the docker client in the Prow container will use the docker daemon in the GCE instance, e.g. for commands like 'docker run ...', 'docker ps ...', etc.

The [recommended way to install docker-machine](https://docs.docker.com/machine/install-machine/) is to curl the binary.